quadsim/robots/robot.py

```diff
@@ -1,44 +1,222 @@
-from typing import Any
-from typing import List
+"""Base class for all robots."""
+import numpy as np
+import pybullet_data
 
-# TODO: move Sensor and Motor to appropriate files
-
-
-class Sensor:
-    pass
-
-
-class Motor:
-    pass
+from quadsim.robots import motor_model
 
 
 class Robot:
     """Robot Base."""
 
-    def __init__(
-        self,
-        pybullet_client: Any,
-        motors: List[Motor],
-        sensors: List[Sensor],
-        on_rack: bool = False,
-    ) -> None:
+    def __init__(self, pybullet_client, config) -> None:
         """Constructs a base robot and resets it to the initial states.
 
         Args:
-
           pybullet_client: The instance of BulletClient to manage different
             simulations.
           motors: A list of motor models.
           sensors: A list of sensor models.
           on_rack: Whether the robot is hanging in mid-air or not. Used for debugging.
         """
+        self._pybullet_client = pybullet_client
+        self.config = config
+        self._setup_simulator()
+        self._load_robot_urdf()
+        self._motor_model = motor_model.MotorModel(self.config.motor)
+        self._num_motors = self._motor_model.num_motors
```

Comment:
It seems redundant to have to manually specify how many motors there are when instantiating a robot when you've already specified this when passing the motor model.

Reply:
Done. Thanks.